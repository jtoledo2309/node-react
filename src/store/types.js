export const AUTH_LOGIN_REQUEST = "AUTH_LOGIN_REQUEST";
export const AUTH_LOGIN_SUCESS = "AUTH_LOGIN_SUCESS";
export const AUTH_LOGIN_FAILURE = "AUTH_LOGIN_FAILURE";

export const AUTH_LOGOUT = "AUTH_LOGOUT";

export const ADVERTS_LOADED = "ADVERTS_LOADED";
export const ADVERT_CREATED = "ADVERT_CREATED";

export const UI_RESET_ERROR = "UI_RESET_ERROR";
