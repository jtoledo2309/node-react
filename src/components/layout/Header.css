.header {
  align-items: center;
  display: flex;
}

.header-logo {
  align-items: center;
  background-color: transparent;
  border-radius: 9999px;
  color: rgb(204, 137, 30);
  display: inline-flex;
  height: 48px;
  justify-content: center;
  transition: background-color 0.2s;
  width: 48px;
}

.header-logo:hover {
  background-color: darkseagreen;
}

.header-nav {
  display: flex;
  flex: 1;
  justify-content: flex-end;
}

.header-nav .active {
  color: rgb(204, 137, 30);
}

.header-button {
  width: 144px;
  margin-left: 10px;
}

.header-addProduct {
  padding: 10px;
}

.header-listProduct {
  padding: 10px;
}
