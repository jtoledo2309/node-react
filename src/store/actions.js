import {
  AUTH_LOGOUT,
  ADVERTS_LOADED,
  AUTH_LOGIN_SUCESS,
  AUTH_LOGIN_REQUEST,
  AUTH_LOGIN_FAILURE,
  UI_RESET_ERROR,
} from "./types.js";

export const authLoginRequest = () => ({
  type: AUTH_LOGIN_REQUEST,
});

export const authLoginSucess = () => ({
  type: AUTH_LOGIN_SUCESS,
});

export const authLoginFailure = (error) => ({
  type: AUTH_LOGIN_FAILURE,
  payload: error,
  error: true,
});

export const authLogout = () => ({
  type: AUTH_LOGOUT,
});

export const advertsLoaded = (adverts) => ({
  type: ADVERTS_LOADED,
  payload: adverts,
});

export const uiResetError = () => ({
  type: UI_RESET_ERROR,
});
