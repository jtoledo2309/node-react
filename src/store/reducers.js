// auth: true/false
// adverts: []
// ui: {
//   isLoading: true/false
//   error: error/null
// }

import {
  AUTH_LOGIN_SUCESS,
  AUTH_LOGOUT,
  ADVERTS_LOADED,
  AUTH_LOGIN_REQUEST,
  AUTH_LOGIN_FAILURE,
  UI_RESET_ERROR,
} from "./types.js";

const defaultState = {
  auth: false,
  adverts: [],
  ui: {
    isLoading: false,
    error: null,
  },
};

// export default function reducer(state = defaultState, action) {
//   switch (action.type) {
//     case AUTH_LOGIN:
//       return { ...state, auth: true };

//     case AUTH_LOGOUT:
//       return { ...state, auth: false };

//     case ADVERTS_LOADED:
//       return { ...state, adverts: action.payload };

//     default:
//       return state;
//   }
// }

export function auth(state = defaultState.auth, action) {
  switch (action.type) {
    case AUTH_LOGIN_SUCESS:
      return true;

    case AUTH_LOGOUT:
      return false;

    default:
      return state;
  }
}

export function adverts(state = defaultState.adverts, action) {
  if (action.type === ADVERTS_LOADED) {
    return action.payload;
  }
  return state;
}

export function ui(state = defaultState.ui, action) {
  if (action.error) {
    return {
      isLoading: false,
      error: action.payload,
    };
  }
  if (/_REQUEST$/.test(action.type)) {
    return {
      error: null,
      isLoading: true,
    };
  }
  if (/SUCESS$/.test(action.type)) {
    return {
      error: null,
      isLoading: false,
    };
  }
  if (action.type === UI_RESET_ERROR) {
    return {
      ...state,
      error: null,
    };
  }
  return state;
}
// switch (action.type) {
//   case AUTH_LOGIN_REQUEST:
//     return {
//       error: null,
//       isLoading: true,
//     };
//   case AUTH_LOGIN_SUCESS:
//     return {
//       error: null,
//       isLoading: false,
//     };
//   case AUTH_LOGIN_FAILURE:
//     return {
//       error: action.payload,
//       isLoading: false,
//     };
//   case UI_RESET_ERROR:
//     return {
//       ...state,
//       error: null,
//     };
//   default:
//     return state;
// }
// export default function reducer(state = defaultState, action) {
//   return {
//     auth: auth(state.auth, action),
//     adverts: adverts(state.adverts, action),
//   };
// }
