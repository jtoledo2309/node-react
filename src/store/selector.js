export const getIsLogged = (state) => state.auth;

export const getAdverts = (state) =>
  state.adverts.areLoaded ? state.adverts.data : [];

export const areAdvertsLoaded = (state) => state.adverts.areLoaded;

export const getUniqueAdvert = (advertId) => (state) =>
  state.adverts.data.find((advert) => advert.id.toString() === advertId);

export const getUi = (state) => state.ui;
