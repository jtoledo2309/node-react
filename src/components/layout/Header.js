import Button from "../common/Button";
import { ReactComponent as Icon } from "../../assets/logo.svg";
import classNames from "classnames";
import { Link, NavLink } from "react-router-dom";
import "./Header.css";
import { logout } from "../auth/service";
import { useAuth } from "../auth/context";

const Header = ({ className }) => {
  const { isLogged, handleLogout } = useAuth();
  const handleLogoutClick = async () => {
    await logout();
    handleLogout();
  };

  return (
    <header className={classNames("header", className)}>
      <Link to="/">
        <div className="header-logo">
          <Icon width="30" height="30" />
        </div>
      </Link>
      <nav className="header-nav">
        <NavLink to="/adverts/new" className="header-addProduct" end>
          AÃ±adir producto{" "}
        </NavLink>
        <NavLink to="/adverts" className="header-listProduct" end>
          Listado de productos
        </NavLink>
        {isLogged ? (
          <Button
            variant="primary"
            className="header-button"
            onClick={handleLogoutClick}
          >
            Log Out
          </Button>
        ) : (
          <Button
            as={Link}
            to="/login"
            variant="primary"
            className="header-button"
          >
            Login
          </Button>
        )}
      </nav>
    </header>
  );
};

export default Header;
