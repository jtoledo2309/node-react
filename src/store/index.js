import { createStore, combineReducers } from "redux";
import { composeWithDevTools } from "@redux-devtools/extension";
import * as reducers from "./reducers";

const reducer = combineReducers(reducers);
export default function configureStore(preloadedState) {
  const store = createStore(reducer, preloadedState, composeWithDevTools());

  return store;
}
