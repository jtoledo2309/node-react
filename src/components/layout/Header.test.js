import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import Header from "./Header";
import { defaultState } from "../../store/reducers";
import { BrowserRouter as Router } from "react-router-dom";

describe("Header", () => {
  test("snapshot", () => {
    const store = {
      getState: () => defaultState,
      dispatch: () => {},
      subscribe: () => {},
    };
    const { container } = render(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
