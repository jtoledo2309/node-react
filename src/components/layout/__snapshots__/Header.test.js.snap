// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <a
      href="/"
    >
      <div
        class="header-logo"
      >
        <svg
          height="30"
          width="30"
        >
          logo.svg
        </svg>
      </div>
    </a>
    <nav
      class="header-nav"
    >
      <a
        class="header-addProduct"
        href="/adverts/new"
      >
        AÃ±adir producto
         
      </a>
      <a
        class="header-listProduct"
        href="/adverts"
      >
        Listado de productos
      </a>
      <a
        class="sc-bcXHqe hQLFWa header-button"
        href="/login"
        variant="primary"
      >
        Login
      </a>
    </nav>
  </header>
</div>
`;
